lib/elixir/lib/enum.ex

```diff
@@ -3355,13 +3355,8 @@ defimpl Enumerable, for: Map do
   end
 
   def reduce(map, acc, fun) do
-    reduce_list(:maps.to_list(map), acc, fun)
+    Enumerable.List.reduce(:maps.to_list(map), acc, fun)
   end
-
-  defp reduce_list(_list, {:halt, acc}, _fun), do: {:halted, acc}
-  defp reduce_list(list, {:suspend, acc}, fun), do: {:suspended, acc, &reduce_list(list, &1, fun)}
-  defp reduce_list([], {:cont, acc}, _fun), do: {:done, acc}
-  defp reduce_list([head | tail], {:cont, acc}, fun), do: reduce_list(tail, fun.(head, acc), fun)
```

Comment:
sad to see this go. I used this source code to learn about implementing `reduce` myself.

Reply:
@sushant12 the same code can be found in `Enumerable.List.reduce/3` :)

https://github.com/elixir-lang/elixir/blob/40756137278cd5ee2e182e86265c21f0a248808d/lib/elixir/lib/enum.ex#L3322-L3325